src/ai/backend/common/validators.py

```diff
@@ -354,6 +364,25 @@ def check_and_return(self, value: Any) -> datetime.tzinfo:
         return tz
 
 
+class TimeDuration(t.Trafaret):
+
+    def check_and_return(self, value: Any) -> datetime.timedelta:
+        if not isinstance(value, str):
+            self._failure('value must be string', value=value)
+        if value[-1] not in ['d', 'h', 'm']:
+            self._failure('value is not a known time duration', value=value)
+        try:
+            t = int(value[:-1])
+            if value[-1] == 'd':
+                return datetime.timedelta(days=t)
+            elif value[-1] == 'h':
+                return datetime.timedelta(hours=t)
+            elif value[-1] == 'm':
+                return datetime.timedelta(days=t * 30)
```

Comment:
The problem of return value error is due to a missing return statement after here, for cases when all of above 3 conditions did not match. If no time unit is specified, we can assume that it's seconds.
```suggestion
                return datetime.timedelta(days=t * 30)
            return datetime.timedelta(seconds=t * 30)
```